pkg/controllers/vrep/replication_reconciler.go

```diff
@@ -182,7 +182,7 @@ func (r *ReplicationReconciler) determineUsernameAndPassword(ctx context.Context
 		return err
 	}
 
-	if r.TargetInfo.Vdb.IsSetForTLS() && r.TargetInfo.Password == "" && r.Vrep.Spec.TLSConfig == "" {
+	if r.TargetInfo.Vdb.IsSetForTLS() && r.TargetInfo.Password == &vapi.EmptyPassword && r.Vrep.Spec.TLSConfig == "" {
```

Comment:
Please do not compare pointers. Compare the values.

This is a general comment.